tests/x11/gedit.pm

```diff
@@ -8,29 +8,21 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# G-Summary: Rework the tests layout.
-# G-Maintainer: Alberto Planas <aplanas@suse.com>
+# Summary: Basic functionality of gedit
+# Maintainer: Oliver Kurz <okurz@suse.de>
 
 use base "x11test";
 use strict;
 use testapi;
 
-# test gedit text editor
-
-# this part contains the steps to run this test
 sub run() {
-    my $self = shift;
+    my ($self) = @_;
     x11_start_program("gedit");
     assert_screen 'gedit-launched';
-    sleep 2;
-    type_string "If you can see this text gedit is working.\n";
-    sleep 2;
-    assert_screen 'test-gedit-1', 3;
-    sleep 2;
-    send_key "alt-f4";
-    sleep 2;
-    send_key "alt-w";
-    sleep 2;
+    $self->enter_test_text('gedit');
+    assert_screen 'test-gedit-1';
+    wait_screen_change { send_key 'alt-f4' };
+    wait_screen_change { send_key 'alt-w' };
```

Comment:
here definitely. Last action doesn't need any waiting - the waiting is implicit in x11test